pandera/backends/pandas/array.py

```diff
@@ -55,11 +55,15 @@ def validate(
         if hasattr(schema, "default") and pd.notna(schema.default):
             check_obj = self.set_default(check_obj, schema)
 
-        if schema.coerce:
-            try:
+        try:
+            if is_field(check_obj) and schema.coerce:
                 check_obj = self.coerce_dtype(check_obj, schema=schema)
-            except SchemaError as exc:
-                error_handler.collect_error(exc.reason_code, exc)
+            elif schema.coerce:
+                check_obj[schema.name] = self.coerce_dtype(
```

Comment:
we should add a test case for this part of the conditional: basically make sure coercing works in the case the `check_obj` is not a column/index